doc/mgmt/Management Framework.md

```diff
@@ -2483,27 +2745,167 @@ Above is the sequence diagram explaining the CVL steps. Note that interaction be
+    6. App module returns the list of keys that it wants to keep a watch on along with the status.
+    7. Translib infra invokes the start transaction request exposed by the DB access layer.
+    8. DB access layer performs a WATCH of all the keys in the Redis DB.
+    9. Status being returned from Redis.
+    10. Status being returned from DB access layer.
+    11. Translib then invokes the prepareUpdate() API on the App module. For Update, Replace, or Delete operations keys are constructed based on the original request keys and existing row entry of corresponding key is fetched from database and the following (table_name, key, row entry, operation) are written in a backup file based on namespace. For Create, only the key and table_name and operation is written in the backup file.     
+        1. Translib invokes the prepareUpdate() API for the keys
+        2. App modules perform operation of taking backup data in the backup file.
+        3. Status being returned from file write.
+        4. Status being returned from App module.
+    12. Translib then invokes the processOperation() API on the App module.
+    13. App modules perform operation of the translated data to the DB access layer.
+    14. DB access layer validates the operation using CVL and then caches them.
+    15. Status being returned from DB access layer.
+    16. Status being returned from App module. 
+17. If operation on all payloads is successful  
+    1. Translib infra invokes the commit transaction on the DB access layer.
+    2. DB access layer first invokes MULTI request on the Redis DB indicating there are multiple writes coming in, so commit everything together. All writes succeed or nothing succeeds.
+    3. Status returned from Redis.
+    4. Pipeline of all the cached writes are executed from the DB access layer.
+    5. Status returned from Redis.
+    6. EXEC call is made to the Redis DB.
+    7. Status returned from Redis DB.
+    8. Status returned from DB access layer
+18. If any operation on payload failed in Step 16. Abort all DB transactions  
+    1. Translib infra invokes the abort transaction on the DB access layer.
+    2. DB access layer first invokes UNWATCH request on the Redis DB indicating abort everything together.
+    3. Status returned from Redis.
+    4. Status returned from DB access layer.     
+19. Post Handling case : Rollback for any failure while commiting
+    Rollback actions will be continued even with failure, however failure during rollback will be logged  
+    1. Non-commited DB Abort: Translib infra invokes the abort transaction on the DB access layer.
+    2. Non-commited DB Abort: DB access layer first invokes UNWATCH request on the Redis DB indicating abort everything together.
+    3. Non-commited DB Abort: Status returned from Redis.
+    4. Non-commited DB Abort: Status returned from DB access layer
+    5. Commited DB Rollback: Revert modified entry by reading backup data from backup file
```

Comment:
This will not work..

1) It creates a churn in the system hw/sw state and can lead to unexpected traffic behavior.
The EXEC at step 17.6 would have resulted in redis notifications to all interested backends and they would have already started acting on it.

2) Since the WATCH scope ends after EXEC, someone could have updated the same entries. This rollback will overwrite them too.